library/ssl_srv.c

```diff
@@ -3107,11 +3107,99 @@ static int ssl_prepare_server_key_exchange( mbedtls_ssl_context *ssl,
             return( ret );
         }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+        if( ciphersuite_info->key_exchange == MBEDTLS_KEY_EXCHANGE_ECDHE_RSA ||
```

Comment:
Minor: I think the call to `mbedtls_ecdh_setup()` above could be in the `else` part of the is branch (that is, skipped if we're taking the PSA path).